roseau/load_flow/models/tests/test_flexible_parameters.py

```diff
@@ -155,8 +171,49 @@ def test_control():
     assert e.value.msg == "'alpha' must be greater than 1 but 0.0 was provided."
     assert e.value.code == RoseauLoadFlowExceptionCode.BAD_CONTROL_VALUE
 
+    # No results to export
+    c = Control(
+        type="q_u",
+        u_min=Q_(210, "V"),
+        u_down=Q_(0.220, "kV"),
+        u_up=Q_(0.230, "kV"),
+        u_max=Q_(2400.5, "dV"),
+    )
+    with pytest.raises(RoseauLoadFlowException) as e:
+        c.results_to_dict()
+    assert e.value.msg == "The Control has no results to export."
+    assert e.value.code == RoseauLoadFlowExceptionCode.JSON_NO_RESULTS
+
+    # From dict with an error (bad control type)
+    c_dict = c.to_dict()
+    c_dict["type"] = "unknown"
+    with pytest.raises(RoseauLoadFlowException) as e:
+        Control.from_dict(c_dict)
+    assert e.value.msg == "Unsupported control type 'unknown'"
+    assert e.value.code == RoseauLoadFlowExceptionCode.BAD_CONTROL_TYPE
+
+    # To dict with an error
+    c.type = "unknown"
```

Comment:
I think `type` should be a property without setter like the other attributes. A `Control` object should be "immutable".